datafusion/core/tests/sql/subqueries.rs

```diff
@@ -21,91 +21,6 @@ use datafusion::assert_batches_eq;
-        "  TableScan: t1 projection=[t1_id] [t1_id:UInt32;N]",
-    ];
-    let formatted = plan.display_indent_schema().to_string();
-    let actual: Vec<&str> = formatted.trim().lines().collect();
-    assert_eq!(
-        expected, actual,
-        "\n\nexpected:\n\n{expected:#?}\nactual:\n\n{actual:#?}\n\n"
-    );
-
-    let sql = "SELECT t1_id from t1 where t1_int = (SELECT t2_int FROM t2 WHERE t2.t2_int = t1.t1_int limit 1)";
-    let msg = format!("Creating logical plan for '{sql}'");
-    let dataframe = ctx.sql(sql).await.expect(&msg);
-    let plan = dataframe.into_optimized_plan()?;
-
-    let expected = vec![
-        "Projection: t1.t1_id [t1_id:UInt32;N]",
-        "  Filter: t1.t1_int = (<subquery>) [t1_id:UInt32;N, t1_int:UInt32;N]",
-        "    Subquery: [t2_int:UInt32;N]",
-        "      Limit: skip=0, fetch=1 [t2_int:UInt32;N]",
-        "        Projection: t2.t2_int [t2_int:UInt32;N]",
-        "          Filter: t2.t2_int = outer_ref(t1.t1_int) [t2_id:UInt32;N, t2_name:Utf8;N, t2_int:UInt32;N]",
-        "            TableScan: t2 [t2_id:UInt32;N, t2_name:Utf8;N, t2_int:UInt32;N]",
-        "    TableScan: t1 projection=[t1_id, t1_int] [t1_id:UInt32;N, t1_int:UInt32;N]",
-    ];
-    let formatted = plan.display_indent_schema().to_string();
-    let actual: Vec<&str> = formatted.trim().lines().collect();
-    assert_eq!(
-        expected, actual,
-        "\n\nexpected:\n\n{expected:#?}\nactual:\n\n{actual:#?}\n\n"
-    );
-
-    let sql = "SELECT t1_id, (SELECT a FROM (select 1 as a) WHERE a = t1.t1_int) as t2_int from t1";
-    let msg = format!("Creating logical plan for '{sql}'");
-    let dataframe = ctx.sql(sql).await.expect(&msg);
-    let plan = dataframe.into_optimized_plan()?;
-
-    let expected = vec![
-        "Projection: t1.t1_id, __scalar_sq_5.a AS t2_int [t1_id:UInt32;N, t2_int:Int64;N]",
-        "  Left Join: CAST(t1.t1_int AS Int64) = __scalar_sq_5.a [t1_id:UInt32;N, t1_int:UInt32;N, a:Int64;N]",
-        "    TableScan: t1 projection=[t1_id, t1_int] [t1_id:UInt32;N, t1_int:UInt32;N]",
-        "    SubqueryAlias: __scalar_sq_5 [a:Int64]",
-        "      Projection: Int64(1) AS a [a:Int64]",
-        "        EmptyRelation []",
-    ];
-    let formatted = plan.display_indent_schema().to_string();
-    let actual: Vec<&str> = formatted.trim().lines().collect();
-    assert_eq!(
-        expected, actual,
-        "\n\nexpected:\n\n{expected:#?}\nactual:\n\n{actual:#?}\n\n"
-    );
-
-    // TODO infer nullability in the schema has bug
-    // // assert data
-    // let results = execute_to_batches(&ctx, sql).await;
-    // let expected = vec![
-    //     "+-------+--------+",
-    //     "| t1_id | t2_int |",
-    //     "+-------+--------+",
-    //     "| 22    |        |",
-    //     "| 33    |        |",
-    //     "| 11    | 1      |",
-    //     "| 44    |        |",
-    //     "+-------+--------+",
-    // ];
-    // assert_batches_eq!(expected, &results);
```

Comment:
Why was this test case commented out even though it's working correctly?
@mingmwang, if you have time, could you please review it? Thank you very much.

Reply:
I remember this was because the infer nullablity has bug causing the physical plan schema check failed. 
I will take a closer look tomorrow, maybe it was already fixed recently.

Reply:
Got it, it should have been fixed. I uncommented the code and everything is working fine.